agent-core/src/main/java/com/newrelic/agent/android/harvest/Harvester.java

```diff
@@ -276,7 +282,24 @@ protected void connected() {
                     break;
             }
 
+            //Offline Storage
+            if (response.isNetworkError()) {
+                checkOfflineAndPersist();
+            }
+
             return;
+        } else {
+            //Offline Storage
```

Comment:
I would move all handling of offline data out of the request method, and probably into into the HarvestLifecycleAware() methods.

Reply:
This part would be better to be here as we need to check the response status.